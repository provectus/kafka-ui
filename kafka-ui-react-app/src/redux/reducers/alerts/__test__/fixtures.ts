export const failurePayload1 = {
  title: 'title',
  message: 'message',
  subject: 'topic-1',
};

export const failurePayload2 = {
  ...failurePayload1,
  subject: 'topic-2',
};
