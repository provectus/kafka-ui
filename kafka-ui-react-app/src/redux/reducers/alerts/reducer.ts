import { getType } from 'typesafe-actions';
import { dismissAlert } from 'redux/actions';
import { Action, AlertsState } from 'redux/interfaces';

import { addError, removeAlert } from './utils';

export const initialState: AlertsState = {};

// eslint-disable-next-line @typescript-eslint/default-param-last
const reducer = (state = initialState, action: Action): AlertsState => {
  const { type } = action;

  if (type.endsWith('__FAILURE')) return addError(state, action);

  if (type === getType(dismissAlert)) {
    return removeAlert(state, action);
  }

  return state;
};

export default reducer;
