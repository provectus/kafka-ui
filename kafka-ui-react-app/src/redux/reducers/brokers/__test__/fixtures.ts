export const brokersPayload = [
  { id: 1, host: 'b-1.test.kafka.amazonaws.com' },
  { id: 2, host: 'b-2.test.kafka.amazonaws.com' },
];

export const brokerStatsPayload = {
  brokerCount: 2,
  zooKeeperStatus: 1,
  activeControllers: 1,
  onlinePartitionCount: 138,
  offlinePartitionCount: 0,
  inSyncReplicasCount: 239,
  outOfSyncReplicasCount: 0,
  underReplicatedPartitionCount: 0,
  diskUsage: [
    { brokerId: 1, segmentSize: 16848434, segmentCount: 118 },
    { brokerId: 2, segmentSize: 12345678, segmentCount: 121 },
  ],
  version: '2.2.1',
};

export const brokersReducerState = {
  items: [],
  brokerCount: 2,
  zooKeeperStatus: 1,
  activeControllers: 1,
  onlinePartitionCount: 138,
  offlinePartitionCount: 0,
  inSyncReplicasCount: 239,
  outOfSyncReplicasCount: 0,
  underReplicatedPartitionCount: 0,
  diskUsage: [
    { brokerId: 1, segmentSize: 16848434, segmentCount: 118 },
    { brokerId: 2, segmentSize: 12345678, segmentCount: 121 },
  ],
  version: '2.2.1',
};
