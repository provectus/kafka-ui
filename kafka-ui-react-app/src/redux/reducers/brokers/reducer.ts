import { Action, BrokersState, ZooKeeperStatus } from 'redux/interfaces';
import { ClusterStats } from 'generated-sources';

export const initialState: BrokersState = {
  items: [],
  brokerCount: 0,
  zooKeeperStatus: ZooKeeperStatus.offline,
  activeControllers: 0,
  onlinePartitionCount: 0,
  offlinePartitionCount: 0,
  inSyncReplicasCount: 0,
  outOfSyncReplicasCount: 0,
  underReplicatedPartitionCount: 0,
  diskUsage: [],
};

const updateBrokerSegmentSize = (
  state: BrokersState,
  payload: ClusterStats
) => {
  const brokers = state.items;
  const { diskUsage } = payload;

  const items = brokers.map((broker) => {
    const brokerMetrics =
      diskUsage && diskUsage.find(({ brokerId }) => brokerId === broker.id);
    if (brokerMetrics !== undefined) {
      return { ...broker, ...brokerMetrics };
    }
    return broker;
  });

  return { ...state, items, ...payload };
};

const reducer = (state = initialState, action: Action): BrokersState => {
  switch (action.type) {
    case 'GET_BROKERS__REQUEST':
      return initialState;
    case 'GET_BROKERS__SUCCESS':
      return {
        ...state,
        items: action.payload,
      };
    case 'GET_CLUSTER_STATUS__SUCCESS':
      return updateBrokerSegmentSize(state, action.payload);
    default:
      return state;
  }
};

export default reducer;
