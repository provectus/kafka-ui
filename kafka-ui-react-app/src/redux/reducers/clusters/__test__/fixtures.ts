import { Cluster, ServerStatus } from 'generated-sources';

export const onlineClusterPayload: Cluster = {
  name: 'secondLocal',
  defaultCluster: true,
  status: ServerStatus.Online,
  brokerCount: 1,
  onlinePartitionCount: 6,
  topicCount: 3,
  bytesInPerSec: 1.55,
  bytesOutPerSec: 9.314,
};
export const offlineClusterPayload: Cluster = {
  name: 'local',
  defaultCluster: false,
  status: ServerStatus.Offline,
  brokerCount: 1,
  onlinePartitionCount: 2,
  topicCount: 2,
  bytesInPerSec: 3.42,
  bytesOutPerSec: 4.14,
};

export const clustersPayload: Cluster[] = [
  onlineClusterPayload,
  offlineClusterPayload,
];
