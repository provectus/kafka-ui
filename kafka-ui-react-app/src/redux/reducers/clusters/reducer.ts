import { Action } from 'redux/interfaces';
import { Cluster } from 'generated-sources';

export const initialState: Cluster[] = [];

const reducer = (state = initialState, action: Action): Cluster[] => {
  switch (action.type) {
    case 'GET_CLUSTERS__SUCCESS':
      return action.payload;
    default:
      return state;
  }
};

export default reducer;
