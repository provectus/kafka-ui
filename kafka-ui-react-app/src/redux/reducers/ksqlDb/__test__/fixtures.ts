export const fetchKsqlDbTablesPayload: {
  tables: Dictionary<string>[];
  streams: Dictionary<string>[];
} = {
  tables: [
    {
      type: 'TABLE',
      name: 'USERS',
      topic: 'users',
      keyFormat: 'KAFKA',
      valueFormat: 'AVRO',
      isWindowed: 'false',
    },
    {
      type: 'TABLE',
      name: 'USERS2',
      topic: 'users',
      keyFormat: 'KAFKA',
      valueFormat: 'AVRO',
      isWindowed: 'false',
    },
  ],
  streams: [
    {
      type: 'STREAM',
      name: 'KSQL_PROCESSING_LOG',
      topic: 'default_ksql_processing_log',
      keyFormat: 'KAFKA',
      valueFormat: 'JSON',
      isWindowed: 'false',
    },
    {
      type: 'STREAM',
      name: 'PAGEVIEWS',
      topic: 'pageviews',
      keyFormat: 'KAFKA',
      valueFormat: 'AVRO',
      isWindowed: 'false',
    },
  ],
};
