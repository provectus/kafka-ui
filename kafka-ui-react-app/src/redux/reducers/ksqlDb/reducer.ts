import { Action } from 'redux/interfaces';
import { getType } from 'typesafe-actions';
import * as actions from 'redux/actions';
import { KsqlState } from 'redux/interfaces/ksqlDb';

export const initialState: KsqlState = {
  streams: [],
  tables: [],
  executionResult: null,
};

// eslint-disable-next-line @typescript-eslint/default-param-last
const reducer = (state = initialState, action: Action): KsqlState => {
  switch (action.type) {
    case getType(actions.fetchKsqlDbTablesAction.success):
      return {
        ...state,
        ...action.payload,
      };
    case getType(actions.executeKsqlAction.success):
      return {
        ...state,
        executionResult: action.payload,
      };
    case getType(actions.resetExecutionResult):
      return {
        ...state,
        executionResult: null,
      };
    default:
      return state;
  }
};

export default reducer;
