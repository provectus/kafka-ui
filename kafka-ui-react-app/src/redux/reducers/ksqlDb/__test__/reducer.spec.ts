import { fetchKsqlDbTablesAction, resetExecutionResult } from 'redux/actions';
import reducer, { initialState } from 'redux/reducers/ksqlDb/reducer';

import { fetchKsqlDbTablesPayload } from './fixtures';

describe('KsqlDb reducer', () => {
  it('returns the initial state', () => {
    expect(reducer(undefined, fetchKsqlDbTablesAction.request())).toEqual(
      initialState
    );
  });
  it('Fetches tables and streams', () => {
    const state = reducer(
      undefined,
      fetchKsqlDbTablesAction.success(fetchKsqlDbTablesPayload)
    );
    expect(state.tables.length).toEqual(2);
    expect(state.streams.length).toEqual(2);
    expect(state).toMatchSnapshot();
  });
  it('Resets execution result', () => {
    const state = reducer(
      {
        tables: [],
        streams: [],
        executionResult: {
          message: 'No available data',
        },
      },
      resetExecutionResult()
    );
    expect(state.executionResult).toEqual(null);
    expect(state).toMatchSnapshot();
  });
});
