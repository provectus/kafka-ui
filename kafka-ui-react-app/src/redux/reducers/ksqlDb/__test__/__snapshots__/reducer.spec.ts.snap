// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KsqlDb reducer Fetches tables and streams 1`] = `
Object {
  "executionResult": null,
  "streams": Array [
    Object {
      "isWindowed": "false",
      "keyFormat": "KAFKA",
      "name": "KSQL_PROCESSING_LOG",
      "topic": "default_ksql_processing_log",
      "type": "STREAM",
      "valueFormat": "JSON",
    },
    Object {
      "isWindowed": "false",
      "keyFormat": "KAFKA",
      "name": "PAGEVIEWS",
      "topic": "pageviews",
      "type": "STREAM",
      "valueFormat": "AVRO",
    },
  ],
  "tables": Array [
    Object {
      "isWindowed": "false",
      "keyFormat": "KAFKA",
      "name": "USERS",
      "topic": "users",
      "type": "TABLE",
      "valueFormat": "AVRO",
    },
    Object {
      "isWindowed": "false",
      "keyFormat": "KAFKA",
      "name": "USERS2",
      "topic": "users",
      "type": "TABLE",
      "valueFormat": "AVRO",
    },
  ],
}
`;

exports[`KsqlDb reducer Resets execution result 1`] = `
Object {
  "executionResult": null,
  "streams": Array [],
  "tables": Array [],
}
`;
